app/models/assignment.rb

```diff
@@ -57,32 +63,35 @@ class Assignment < Assessment
   accepts_nested_attributes_for :assignment_files, allow_destroy: true
   validates_associated :assignment_files
 
-  # this has to be before :peer_reviews or it throws a HasManyThroughOrderError
-  has_many :groupings, foreign_key: :assessment_id, inverse_of: :assignment
   # Assignments can now refer to themselves, where this is null if there
   # is no parent (the same holds for the child peer reviews)
   belongs_to :parent_assignment,
              class_name: 'Assignment', optional: true, inverse_of: :pr_assignment, foreign_key: :parent_assessment_id
-  has_one :pr_assignment, class_name: 'Assignment', foreign_key: :parent_assessment_id, inverse_of: :parent_assignment
-  has_many :peer_reviews, through: :groupings
-  has_many :pr_peer_reviews, through: :parent_assignment, source: :peer_reviews
+  has_one :pr_assignment,
+          class_name: 'Assignment',
+          dependent: :destroy,
+          foreign_key: :parent_assessment_id,
+          inverse_of: :parent_assignment
+  has_many :peer_reviews, dependent: :destroy, through: :groupings
+  has_many :pr_peer_reviews, through: :parent_assignment, source: :peer_reviews, dependent: :destroy
 
-  has_many :current_submissions_used, through: :groupings,
-                                      source: :current_submission_used
+  has_many :current_submissions_used,
+           through: :groupings,
+           source: :current_submission_used,
+           dependent: :destroy
```

Comment:
Don't add this to `through` associations (it's either redundant or misleading: in this case, for example, `groupings` can't be destroyed, and so `current_submissions_used` will also never be destroyed)